app/services/island_service.rb

```diff
@@ -0,0 +1,57 @@
+class IslandService
+
+  ITERATIONS = 8
+
+  def initialize(position, size)
+    @position = position
+    @size = size
+
+  end
+
+  def call
+    island = create_polygon
+  end
+
+  def generate_midpoint(one_edge)
+    pt1 = one_edge.first
+    pt2 = one_edge.last
+    mx = (pt1.first + pt2.first) / 2
+    my = (pt1.last + pt2.last) / 2
+    vx = -(pt1.last - pt2.last)
+    vy =  (pt1.first - pt2.first)
+    d = [-1, 1].sample * Random.rand/2
+    [(mx + (d * vx)),(my + (d * vy))]
+    #[mx,my]
+  end
+
+  def generate_midpoints(edges)
+    edges.map do |one_edge|
+      if distance(one_edge) < 5
+        [one_edge.first]
+      else
+      [one_edge.first , generate_midpoint(one_edge)]
+      end
+    end
```

Comment:
the `flatten(1)` from below should be here in order for the function to return something meaningful